static/app/views/ddm/useGetChartPalette.tsx

```diff
@@ -0,0 +1,62 @@
+import {useCallback, useRef} from 'react';
+
+import theme from 'sentry/utils/theme';
+
+const CACHE_SIZE = 20; // number of palettes to cache
+
+// We do length - 2 to be aligned with the colors in other parts of the app (copy-pasta)
+// We use Math.max to avoid numbers < -1 as then `getColorPalette` returns undefined (not typesafe because of array access)
+export function getChartColors(seriesCount: number) {
+  return theme.charts.getColorPalette(Math.max(seriesCount - 2, -1));
```

Comment:
If you want to be safer here, you can call getColorPalette and default to one of the palette colors directly
```suggestion
  return theme.charts.getColorPalette(seriesCount - 2) ?? CHART_PALETTE[some default index for palette],
```